meta/meta.go

```diff
@@ -488,6 +491,73 @@ func (m *Meta) CreateTableOrView(dbID int64, tableInfo *model.TableInfo) error {
 	return m.txn.HSet(dbKey, tableKey, data)
 }
 
+// SetDDLTables write a key into storage.
+func (m *Meta) SetDDLTables() error {
+	_, err := m.txn.Inc(mInitDDLTable, 1)
+	return errors.Trace(err)
+}
+
+// CreateMySQLSchema create mysql schema
+func (m *Meta) CreateMySQLSchema() (int64, error) {
+	dbs, err := m.ListDatabases()
+	if err != nil {
+		return 0, err
+	}
+	for _, db := range dbs {
+		if db.Name.L == mysql.SystemDB {
+			return db.ID, nil
+		}
+	}
+
+	id, err := m.GenGlobalID()
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	db := model.DBInfo{
+		ID:      id,
+		Name:    model.NewCIStr(mysql.SystemDB),
+		Charset: mysql.UTF8MB4Charset,
+		Collate: mysql.UTF8MB4DefaultCollation,
+		State:   model.StatePublic,
+	}
+	data, err := json.Marshal(db)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+
+	return db.ID, m.txn.HSet(mDBs, m.dbKey(db.ID), data)
+}
+
+// CheckDDLTableExists check if the tables related to concurrent DDL exists.
+func (m *Meta) CheckDDLTableExists() (bool, error) {
+	v, err := m.txn.Get(mInitDDLTable)
```

Comment:
Why do we need another new meta key?

How can we handle the inconsistency between `initDDLTable` and the existences of `mysql.tidb_ddl_job`? For example, if a user mistakenly delete the table `mysql.tidb_ddl_job`, the cluster is inavailable.

Reply:
there are many other important tables in `mysql`, and after drop them tidb is unavailable too. The user should not expect tidb is still healthy if they drop a system table